internal/host/hostcommands/instruction_manager.go

```diff
@@ -136,11 +144,20 @@ func (i *InstructionManager) GetNextSteps(ctx context.Context, host *models.Host
 			if steps == nil {
 				continue
 			}
+			// Creating StepID when needed and filtering out disabled steps
+			idx := 0
 			for _, step := range steps {
+				if i.isStepDisabled(step.StepType) {
+					log.WithField("disabledStepsMap", i.disabledStepsMap).Infof("Step '%v' is disabled. Will not include it in instructions", step.StepType)
+					continue
+				}
 				if step.StepID == "" {
 					step.StepID = createStepID(step.StepType)
 				}
+				steps[idx] = step
+				idx++
 			}
+			steps = steps[:idx]
```

Comment:
So you override the same list you are iterating.. This behavior is both confusing and error-prone.
I'd suggest appending to a new one `filteredSteps`

Reply:
In this case, I don't see that it can cause errors since we always modify elements in indexes that we already passed, And never change size of the array until after the loop. But it might be confusing. Ill update.